addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -135,7 +135,9 @@ const Link = Widget.extend({
                 }
                 active = subActive;
             } else {
-                active = !this.data.iniClassName || this.data.iniClassName.includes('btn-link') || !this.data.iniClassName.includes('btn-');
+                const supportedTypes = this.colorsData.map(c => c.type);
+                active = !this.data.iniClassName
+                    || !this.data.iniClassName.split(' ').find(el => supportedTypes.includes(el));
```

Comment:
Unless I am missing something, `supportedTypes` contains either `'link', 'primary', 'secondary', 'custom'` or `'', 'primary', 'secondary', 'custom'`, depending on if it's a button or a link we edit. You then loop through classes and try to find one that matches one of those "supported types" which are not class names :shrug: 

What I am not understanding right?